examples/tutorials/advanced/legends.py

```diff
@@ -0,0 +1,141 @@
+"""
+Creating legends
+================
+
+The :meth:`pygmt.Figure.legend` method creates legends, whereby auto-legends as well
+as manually created legends are supported.
+"""
+
+# %%
+import io
+
+import pygmt
+
+# %%
+# Create an auto-legend
+# ---------------------
+#
+# For auto-legends, the ``label`` parameter of :meth:`pygmt.Figure.plot` has to be
+# specified to state the desired text in the legend entry (white spaces are allowed).
+# Optionally, to adjust the legend, users can append different modifiers to the string
+# passed to ``label``. A list of all available modifiers can be found at
+# :gmt-docs:`gmt.html#l-full`. To create a
+# :doc:`multiple-column legend </gallery/embellishments/legend>` **+N** is used with
+# the desired number of columns. By default, the legend is placed in the Upper Right
+# corner with an offset of 0.1 centimeters in both x and y directions and a box with
+# a white fill and a 1-point thick, black, solid outline is drawn around the legend.
+
+fig = pygmt.Figure()
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+
+fig.plot(x=0, y=0, style="c0.25c", fill="orange", label="orange circle")
```

Comment:
Better to call `Figure.plot` twice with different labels, so that users can know the default layout for legend entries.

Reply:
Hm. Now, three symbols are plotted to show that by default a one-column legend is created and the order of the symbols is given by the plotting order. Or did you mean something different?